mongodbatlas/fw_resource_mongodbatlas_search_node.go

```diff
@@ -0,0 +1,361 @@
+	if _, _, err := connV2.AtlasSearchApi.UpdateAtlasSearchDeployment(ctx, projectID, clusterName, &searchDeploymentReq).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during search node update", err.Error())
+		return
+	}
+
+	updateTimeout, diags := searchNodePlan.Timeouts.Update(ctx, defaultSearchNodeTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	deploymentResp, err := waitSearchNodeStateTransition(ctx, projectID, clusterName, connV2, updateTimeout)
+	if err != nil {
+		resp.Diagnostics.AddError("error during search node update", err.Error())
+		return
+	}
+	newSearchNodeModel, diagnostics := newTFSearchDeployment(ctx, clusterName, deploymentResp, searchNodePlan.Timeouts)
+	resp.Diagnostics.Append(diagnostics...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newSearchNodeModel)...)
+}
+
+func (r *SearchNodeRS) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var searchNodeState *tfSearchNodeRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &searchNodeState)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.client.AtlasV2
+	projectID := searchNodeState.ProjectID.ValueString()
+	clusterName := searchNodeState.ClusterName.ValueString()
+	if _, err := connV2.AtlasSearchApi.DeleteAtlasSearchDeployment(ctx, projectID, clusterName).Execute(); err != nil {
+		resp.Diagnostics.AddError("error during search node delete", err.Error())
+		return
+	}
+
+	deleteTimeout, diags := searchNodeState.Timeouts.Delete(ctx, defaultSearchNodeTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	if err := waitSearchNodeDelete(ctx, projectID, clusterName, connV2, deleteTimeout); err != nil {
+		resp.Diagnostics.AddError("error during search node delete", err.Error())
+		return
+	}
+}
+
+func (r *SearchNodeRS) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	projectID, clusterName, err := splitSearchNodeImportID(req.ID)
+	if err != nil {
+		resp.Diagnostics.AddError("error splitting search node import ID", err.Error())
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectID)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("cluster_name"), clusterName)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func splitSearchNodeImportID(id string) (projectID, clusterName string, err error) {
+	var re = regexp.MustCompile(`(?s)^([0-9a-fA-F]{24})-(.*)$`)
+	parts := re.FindStringSubmatch(id)
+
+	if len(parts) != 3 {
+		err = errors.New("use the format {project_id}-{cluster_name}")
+		return
+	}
+
+	projectID = parts[1]
+	clusterName = parts[2]
+	return
+}
+
+func waitSearchNodeStateTransition(ctx context.Context, projectID, clusterName string, client *admin.APIClient, timeout time.Duration) (*admin.ApiSearchDeploymentResponse, error) {
```

Comment:
could we perhaps start considering having separate files within the same package that contain these wait search methods?

Reply:
I agree this is a good example of set of functions that can be in a separate file when having resource implementations separated in packages/folders.